aderyn_core/src/context/graph/workspace_callgraph.rs

```diff
@@ -0,0 +1,134 @@
+use std::collections::{hash_map, HashMap, HashSet};
```

Comment:
This is a big chunk of potentially _very_ impactful code. Could you please add some unit tests for each of the functions here to ensure the callgraph is being created correctly?